presto-spi/src/main/java/com/facebook/presto/spi/prerequisites/QueryPrerequisitesContext.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.prerequisites;
+
+import com.facebook.presto.spi.ConnectorId;
+
+import java.util.Map;
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+
+public class QueryPrerequisitesContext
+{
+    private final Optional<String> catalog;
+    private final Optional<String> schema;
+    private final String query;
+    private final Map<String, String> systemProperties;
+    private final Map<ConnectorId, Map<String, String>> connectorProperties;
+
+    public QueryPrerequisitesContext(
```

Comment:
Aren't all of these already in Session? Maybe add a convenience constructor that takes Session object so that it's easier to use this class.

Reply:
`Session` is not a part of SPI and hence I cannot use it. Even I would have preferred it instead of making a new one